### PR TITLE
synchronize with drcsim plugins

### PR DESCRIPTION
**Pull request by hsu**

### CODE REVIEW

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.h [hunk 1]
```diff
@@ -1,39 +1,32 @@
...
- *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  *
- */
-/*
- * Desc: A dynamic controller plugin that publishes ROS image_raw camera_info topic for generic camera sensor.
```
Comment (davetcoleman) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.h:22]: we should keep this description

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera_utils.h [hunk 2]
```diff
@@ -83,15 +82,17 @@
...
+    /// \brief A mutex to lock access to image_connect_count_
+    protected: boost::shared_ptr<boost::mutex> image_connect_count_lock_;
     protected: void ImageConnect();
     protected: void ImageDisconnect();
 
     /// \brief Keep track when we activate this camera through ros
     /// subscription, was it already active?  resume state when
     /// unsubscribed.
-    protected: bool was_active_;
+    protected: boost::shared_ptr<bool> was_active_;
```
Comment (davetcoleman) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera_utils.h:95]: why a shared pointer for a bool??

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_force.h [hunk 3]
```diff
@@ -22,49 +22,45 @@
...
 #include <boost/thread/mutex.hpp>
 
-#include <gazebo/physics/physics.hh>
-#include <gazebo/transport/TransportTypes.hh>
-#include <gazebo/common/Plugin.hh>
-#include <gazebo/common/Events.hh>
+#include "gazebo/physics/physics.hh"
+#include "gazebo/transport/TransportTypes.hh"
+#include "gazebo/common/Plugin.hh"
+#include "gazebo/common/Events.hh"
```
Comment (davetcoleman) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_force.h]: We want to keep the brackets

File: gazebo_plugins/src/gazebo_ros_camera.cpp [hunk 4]
```diff
@@ -22,7 +22,9 @@
    Date: 24 Sept 2008
 */
 
-#include <gazebo_plugins/gazebo_ros_camera.h>
+#include "gazebo_plugins/gazebo_ros_camera.h"
```
Comment (davetcoleman) [inline gazebo_plugins/src/gazebo_ros_camera.cpp:25]: brackets

File: gazebo_plugins/src/gazebo_ros_gpu_laser.cpp [hunk 5]
```diff
@@ -25,18 +25,18 @@
...
-#include <sdf/sdf.hh>
-#include <sdf/Param.hh>
 #include <gazebo/common/Exception.hh>
 #include <gazebo/sensors/GpuRaySensor.hh>
 #include <gazebo/sensors/SensorTypes.hh>
 
 #include <tf/tf.h>
 
-#include <gazebo_plugins/gazebo_ros_gpu_laser.h>
+#include "gazebo_plugins/gazebo_ros_gpu_laser.h"
```
Comment (davetcoleman) [inline gazebo_plugins/src/gazebo_ros_gpu_laser.cpp:39]: brackets

File: gazebo_plugins/src/gazebo_ros_p3d.cpp [hunk 6]
```diff
@@ -15,17 +15,11 @@
  *
 */
 
-/*
- * Desc: 3D position interface for ground truth.
```
Comment (davetcoleman) [inline gazebo_plugins/src/gazebo_ros_p3d.cpp:19]: why is this removed?

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera_utils.h (hunk 2 above)
Comment (hsu) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera_utils.h:95]: see [this pull request](https://bitbucket.org/osrf/drcsim/pull-request/332/fixes-issue-217-camera-subscription/diff)

File: gazebo_plugins/src/gazebo_ros_camera.cpp (hunk 4 above)
Comment (hsu) [inline gazebo_plugins/src/gazebo_ros_camera.cpp:25]: this one is local right?

File: gazebo_plugins/src/gazebo_ros_gpu_laser.cpp (hunk 5 above)
Comment (hsu) [inline gazebo_plugins/src/gazebo_ros_gpu_laser.cpp:39]: changed all headers in `gazebo_plugin` to use `#include` with brackets.  Left `*.cpp` with quotes given people should not build against cpp files.

File: gazebo_plugins/src/gazebo_ros_p3d.cpp (hunk 6 above)
Comment (hsu) [inline gazebo_plugins/src/gazebo_ros_p3d.cpp:19]: added back in header

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_force.h (hunk 3 above)
Comment (hsu) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_force.h]: recovered.

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.h (hunk 1 above)
Comment (hsu) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.h:22]: added back

File: gazebo_plugins/src/gazebo_ros_camera.cpp (hunk 4 above)
Comment (hsu) [inline gazebo_plugins/src/gazebo_ros_camera.cpp:25]: changed all headers in `gazebo_plugin` to use `#include` with brackets.  Left `*.cpp` with quotes given people should not build against cpp files.

**Comment by davetcoleman** [comment]
+1
